portalnet/src/overlay.rs

```diff
@@ -535,20 +543,16 @@ where
         let cid = utp_rs::cid::ConnectionId {
             recv: conn_id,
             send: conn_id.wrapping_add(1),
-            peer: crate::discovery::UtpEnr(enr),
+            peer: UtpEnr(enr),
         };
-        let mut stream = self
-            .utp_controller
-            .connect_with_cid(cid, *UTP_CONN_CFG)
-            .await
-            .map_err(|err| OverlayRequestError::UtpError(format!("{err:?}")))?;
-        let mut data = vec![];
-        stream
-            .read_to_eof(&mut data)
+        self.utp_controller
+            .connect_inbound_stream(cid)
             .await
-            .map_err(|err| OverlayRequestError::UtpError(format!("{err:?}")))?;
-
-        Ok(data)
+            .map_err(|err| OverlayRequestError::ContentNotFound {
+                message: format!("Unable to locate content on the network: {err:?}"),
+                utp: true,
+                trace: None,
+            })
```

Comment:
we already through this error in the utp_controller no? so we can probably remove this `map_err`